core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -83,4 +85,25 @@ protected void deleteFiles(Set<String> pathsToDelete, String fileType) {
             (file, thrown) -> LOG.warn("Delete failed for {} file: {}", fileType, file, thrown))
         .run(deleteFunc::accept);
   }
+
+  protected Set<String> expiredStatisticsFilesLocations(
+      TableMetadata beforeExpiration, TableMetadata afterExpiration) {
+    Set<String> statisticsFilesLocationBeforeExpiry = allStatisticsFilesLocation(beforeExpiration);
+    Set<String> statisticsFilesLocationAfterExpiry = allStatisticsFilesLocation(afterExpiration);
+    statisticsFilesLocationBeforeExpiry.removeAll(statisticsFilesLocationAfterExpiry);
+    return statisticsFilesLocationBeforeExpiry;
+  }
+
+  private Set<String> allStatisticsFilesLocation(TableMetadata tableMetadata) {
```

Comment:
How about `statsFileLocations` instead? Since it returns a collection, the method name should be plural. And we also avoid needless modifiers, like `all` (which is assumed).